viziphant/unitary_event_analysis.py

```diff
@@ -0,0 +1,401 @@
+                       ls='None', marker='.', markersize=0.5, color='k')
+            indices_of_significant_joint_surprises = np.where(
+                joint_surprise_dict['Js'] >= joint_surprise_significance)[0]
+            if len(indices_of_significant_joint_surprises) > 0:
+                indices_of_coincidence_events = np.unique(
+                    joint_surprise_dict['indices']['trial' + str(trial)])
+                if len(indices_of_coincidence_events) > 0:
+                    indices_of_unitary_events = []
+                    for j in indices_of_significant_joint_surprises:
+                        coincidence_indices_greater_left_window_margin = \
+                            indices_of_coincidence_events * binsize >= \
+                            t_winpos[j]
+                        coincidence_indices_smaller_right_window_margin = \
+                            indices_of_coincidence_events * binsize < \
+                            t_winpos[j] + window_size
+                        coincidence_indices_in_actual_analysis_window = \
+                            coincidence_indices_greater_left_window_margin & \
+                            coincidence_indices_smaller_right_window_margin
+                        indices_of_unitary_events = \
+                            np.append(
+                                indices_of_unitary_events,
+                                indices_of_coincidence_events
+                                [coincidence_indices_in_actual_analysis_window]
+                                )
+                    unitary_events_on_timescale = \
+                        np.unique(indices_of_unitary_events) * binsize
+                    unitary_events_on_trialscale = \
+                        np.ones_like(np.unique(indices_of_unitary_events)) * \
+                        trial + n * (n_trials + 1) + 1
+                    axis6.plot(unitary_events_on_timescale,
+                               unitary_events_on_trialscale,
+                               markersize=params_dict['marker_size'],
+                               marker='s', ls='', markerfacecolor='none',
+                               markeredgecolor='r')
+        if n < n_neurons - 1:
+            axis6.axhline((trial + 2) * (n + 1), lw=params_dict['lw'],
+                          color='k')
+    axis6.set_xlim(xlim_left, xlim_right)
+    axis6.set_ylim(0, (trial + 2) * (n + 1) + 1)
+    axis6.xaxis.set_major_locator(MaxNLocator(integer=True))
+    axis6.set_yticks(y_ticks_list)
+    axis6.set_yticklabels(y_ticks_labels_list, fontsize=params_dict['fsize'])
+    axis6.set_xlabel('Time (ms)', fontsize=params_dict['fsize'])
+    axis6.set_ylabel('Trial', fontsize=params_dict['fsize'])
+
+    # mark all epochs on all subplots and annotate all subplotaxis
+    for n in range(1, 7):
+        mark_epochs(eval(f"axis{n}"))
+        axis = locals()['axis'+str(n)]
+        axis.text(-0.05, 1.1, string.ascii_uppercase[n-1],
```

Comment:
Explicit is better than implicit. Consider explicitly adding the letters next to plots.

Reply:
do you mean like: (e.g)
axes1.text(-0.05, 1.1, 'A',
                  transform=axes.transAxes, size=params_dict['fsize'] + 5,
                  weight='bold')

Reply:
Something like this:
```
axes_list = [axes1, axes2, axes3, axes4, axes5, axes6]
letter_list = ['A', 'B', 'C', 'D', 'E', 'F']
for (axes, letter) in zip(axes_list, letter_list):
    axes.text(-0.05, 1.1, letter, transform=axes.transAxes, size=params_dict['fsize'] + 5,
                    weight='bold')
```
However, please do not just copy-paste the lines above as it is untested. Also, Danylo, feel free to comment on the approach.